### PR TITLE
feat(TDOPS-940/ObjectViewer): add hideTooltip props

### PR DESCRIPTION
**Pull request by hbhong**
**What is the problem this PR is trying to solve?**
Supports hiding tooltip for ObjectViewer(JSONLike) component

**What is the chosen solution to this problem?**
A new prop "hideTooltip" is added.

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/ObjectViewer/JSONLike/JSONLike.component.js [hunk 1]
```diff
@@ -368,7 +363,26 @@ function UntranslatedComplexItem(props) {
...
+							key="badge"
+							className={`${theme.badge} badge`}
+							aria-label={t('TC_OBJECT_VIEWER_NB_CHILD', {
+								defaultValue: 'Contains {{count}} child object',
+								defaultValue_plural: 'Contains {{count}} child objects',
+								count: info.length,
+							})}
+						>
+							{decoratedLength}
+						</sup>
```
Comment (jmfrancois) [inline packages/components/src/ObjectViewer/JSONLike/JSONLike.component.js:383]: please factorize it to not copy pasate those 10 lines

File: packages/components/src/ObjectViewer/JSONLike/JSONLike.test.js [hunk 2]
```diff
@@ -44,6 +44,30 @@ describe('JSONLike', () => {
 		expect(wrapper.props().className).toContain('extra-test');
 	});
 
+	it('should render with tooltip', () => {
```
Comment (jmfrancois) [inline packages/components/src/ObjectViewer/JSONLike/JSONLike.test.js:47]:
```suggestion
	it('should render without tooltip', () => {
```

File: packages/components/src/ObjectViewer/JSONLike/JSONLike.test.js [hunk 3]
```diff
@@ -184,6 +208,22 @@ describe('JSONLike', () => {
 			// expect
 			expect(wrapper.getElement()).toMatchSnapshot();
 		});
+		it('should render without tooltip', () => {
+			// when
+			const wrapper = shallow(
```
Comment (jmfrancois) [inline packages/components/src/ObjectViewer/JSONLike/JSONLike.test.js:213]: please  use react testing library. enzyme is deprecated and block us to update to react 18

**Review by jmfrancois — CHANGES_REQUESTED** [review]

File: packages/components/src/ObjectViewer/JSONLike/JSONLike.component.js (hunk 1 above)
Comment (hbhong) [reply packages/components/src/ObjectViewer/JSONLike/JSONLike.component.js]: Thanks, fixed it.

File: packages/components/src/ObjectViewer/JSONLike/JSONLike.test.js (hunk 3 above)
Comment (hbhong) [reply packages/components/src/ObjectViewer/JSONLike/JSONLike.test.js]: Thanks, I updated the test cases with RTL.

File: packages/components/src/ObjectViewer/JSONLike/JSONLike.component.js [hunk 4]
```diff
@@ -311,6 +311,7 @@ function UntranslatedComplexItem(props) {
 				{info.keys.map((key, i) => {
 					const childId = id && `${id}-${key}`;
 					return (
+						// eslint-disable-next-line @typescript-eslint/no-use-before-define
```
Comment (hbhong) [inline packages/components/src/ObjectViewer/JSONLike/JSONLike.component.js:314]: it's a circular dependency between the Item and ComplexItem,so disable the eslint check.

**Review by jmfrancois — APPROVED** [review]

File: packages/components/src/ObjectViewer/JSONLike/JSONLike.component.js (hunk 4 above)
Comment (jmfrancois) [reply packages/components/src/ObjectViewer/JSONLike/JSONLike.component.js:314]: :sweat_smile:

File: packages/components/src/ObjectViewer/JSONLike/JSONLike.test.js (hunk 3 above)
Comment (jmfrancois) [reply packages/components/src/ObjectViewer/JSONLike/JSONLike.test.js]: Thanks !

**Review by ybaskaran — APPROVED** [review]

File: packages/components/src/ObjectViewer/JSONLike/JSONLike.component.js [hunk 5]
```diff
@@ -327,7 +328,19 @@ function UntranslatedComplexItem(props) {
 			</ul>
 		);
 	}
-
+	const childCount = (
```
Comment (ybaskaran) [inline packages/components/src/ObjectViewer/JSONLike/JSONLike.component.js:331]: [Optional] can be named as childCountInfo

Comment (ybaskaran) [reply packages/components/src/ObjectViewer/JSONLike/JSONLike.component.js:331]: childCount seems more like a integer value

File: packages/components/src/ObjectViewer/JSONLike/JSONLike.component.js [hunk 6]
```diff
@@ -534,7 +542,9 @@ export function JSONLike({ onSubmit, className, style, ...props }) {
 	if (rootIsObject) {
 		if (props.rootLabel) {
 			labelId = (props.id && `${props.id}-label`) || 'tc-object-viewer-label';
-			label = (
+			label = props.hideTooltip ? (
+				<div className={theme['root-label-overflow']}>{props.rootLabel}</div>
```
Comment (ybaskaran) [inline packages/components/src/ObjectViewer/JSONLike/JSONLike.component.js:546]: [Optional]
used twice so may be we can have a const,
const labelText = <div className={theme['root-label-overflow']}>{props.rootLabel}</div>;

**Pull request merged**